app/controllers/admin/grove_monitor_controller.rb

```diff
@@ -1,5 +1,12 @@
 class Admin::GroveMonitorController < ApplicationController
   def show
-    authorize(:student_management, :show?)
+    @students = current_user.grove.students.send(filter_params)
```

Comment:
[law of demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) says `foo.bar.baz` is a bad idea since you're reaching across object boundaries. And, you can probably do a more performant query if you have a `Teacher.students_in_grove` scope (or a helper class that contextually does the query) that just does a `Student.where(grove_id: self.grove_id)` not really sure how the send is working here, but i trust it does.